### PR TITLE
[CGP-400] Use bytestrings and proper hashes

### PR DESCRIPTION
**Pull request by j-mueller**
* Previously the `PendingTx` type (representing the transaction that is
  being validated in a PLC validator script) used `Int`s instead of
  `ByteString`s for hashes
* With this commit we use the correct `ByteString` type and get all
  hashes from the mockchain, instead of having to use stand-in values
* Note that this also changes the UTXO model in `Wallet.UTXO.Types`:
  The `PayToScript` constructor of `TxOutType` now contains the hash of
  the validator script as a field, where previously that hash was included
  implicitly, in the address generated by `scriptTxOut` in the same
  module.
  The address is a combined hash of value, validator and data script.
  However, if we want to ensure that transaction outputs are locked by a
  specific validator script, we need to look at the hash of that script
  on its own.
  (see plutus-use-cases/Language/Plutus/Coordination/Contracts/Vesting.hs,
  l. 158 for an example)

### CODE REVIEW

File: plutus-use-cases/plutus-use-cases.cabal [hunk 1]
```diff
@@ -69,7 +69,8 @@ test-suite plutus-use-cases-test
         plutus-th -any,
         microlens -any,
         core-to-plc -any,
-        template-haskell -any
+        template-haskell -any,
+        bytestring -any
```
Comment (j-mueller) [inline plutus-use-cases/plutus-use-cases.cabal]: Actually I think `bytestring` isn't necessary here

File: wallet-api/src/Wallet/UTXO/Runtime.hs [hunk 2]
```diff
@@ -33,9 +39,9 @@ instance LiftPlc PendingTxOutType
 
 -- | Output of a pending transaction.
 data PendingTxOut = PendingTxOut {
-    pendingTxOutValue   :: Value,
-    pendingTxDataScript :: Maybe Hash,
-    pendingTxOutData    :: PendingTxOutType
+    pendingTxOutValue  :: Value,
+    pendingTxOutHashes :: Maybe (Hash, Hash), -- ^ Hashes of validator script and data script
```
Comment (michaelpj) [inline wallet-api/src/Wallet/UTXO/Runtime.hs]: Could we type this a bit better so it's not so easy to mix up the two hashes? Likewise below.

**Comment by michaelpj** [comment]
I think the digest-vs-hash thing deserves a note in the code. And maybe we should call `Hash` something else, the current name is confusingly generic.

**Comment by j-mueller** [comment]
> I think the digest-vs-hash thing deserves a note in the code. 

I'll add that

>And maybe we should call Hash something else, the current name is confusingly generic.

Just added a commit that replaces `Hash` with more specific newtypes

**Comment by michaelpj** [comment]
LGTM

**Pull request merged**